spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/config/KinesisBinderConfiguration.java

```diff
@@ -64,17 +66,40 @@ public KinesisStreamProvisioner provisioningProvider(AmazonKinesisAsync amazonKi
 	}
 
 	@Bean
-	KinesisMessageChannelBinder kinesisMessageChannelBinder(
-			AmazonKinesisAsync amazonKinesis,
-			KinesisStreamProvisioner provisioningProvider,
-			Codec codec) {
+	KinesisMessageChannelBinder kinesisMessageChannelBinder(AmazonKinesisAsync amazonKinesis,
+			KinesisStreamProvisioner provisioningProvider, Codec codec, MetadataStore kinesisCheckpointStore) {
 
-		KinesisMessageChannelBinder kinesisMessageChannelBinder =
-				new KinesisMessageChannelBinder(amazonKinesis, this.configurationProperties, provisioningProvider);
+		KinesisMessageChannelBinder kinesisMessageChannelBinder = new KinesisMessageChannelBinder(amazonKinesis,
+				this.configurationProperties, provisioningProvider);
 		kinesisMessageChannelBinder.setCodec(codec);
+		kinesisMessageChannelBinder.setMetadataStore(kinesisCheckpointStore);
 
 		return kinesisMessageChannelBinder;
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	MetadataStore kinesisCheckpointStore(AWSCredentialsProvider awsCredentialsProvider, RegionProvider regionProvider) {
+
+		String tableName = this.configurationProperties.getCheckpointTable();
+
+		MetadataStore kinesisCheckpointStore;
+		if (tableName != null) {
+			AmazonDynamoDBAsync dynamoDB = AmazonDynamoDBAsyncClientBuilder.standard()
+					.withCredentials(awsCredentialsProvider)
+					.withRegion(regionProvider.getRegion().getName())
+					.build();
+
+			kinesisCheckpointStore = new DynamoDbMetaDataStore(dynamoDB, tableName);
+			((DynamoDbMetaDataStore) kinesisCheckpointStore)
+					.setReadCapacity(configurationProperties.getDynamoDbReadCapacity());
+			((DynamoDbMetaDataStore) kinesisCheckpointStore)
+					.setWriteCapacity(configurationProperties.getDynamoDbWriteCapacity());
+		} else {
+			kinesisCheckpointStore = new SimpleMetadataStore();
```

Comment:
Why do you keep this here?
I made a request:

> In my opinion the Kinesis Binder does not make sense out side of AWS, so that definitely feels organic to always rely on the DynamoDBMetadataStore with some default table name. If you somehow need to use some other MetadataStore impl, you always can specify a @Bean for it. For example that was my idea for tests scope in the project. Of course checkpointTable property will be ignored.

and it hasn't been addressed either way: nor fix, not even comment.

I might won't mind to keep your code as is but I need strong reason.
Thanks